docusaurus/docs/create-a-plugin/extend-a-plugin/nested-plugins.md

```diff
@@ -0,0 +1,87 @@
+---
+id: nested-plugins
+title: Work with nested plugins
+description: How to work with nested plugins, Grafana app plugins.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - advanced
+  - apps
+  - app plugins
+  - nested
+---
+
+import ScaffoldNPM from '@snippets/createplugin-scaffold.npm.md';
+import ScaffoldPNPM from '@snippets/createplugin-scaffold.pnpm.md';
+import ScaffoldYarn from '@snippets/createplugin-scaffold.yarn.md';
+
+Grafana app plugins can nest data sources, both frontend and backend, together with panel plugins so that you can provide a complete user experience.
+
+## Before you begin
+
+See the tutorial to [build an app plugin](../../tutorials/build-an-app-plugin) and scaffold an app plugin.
+
+## Anatomy of nested plugins
+
+Nested plugins are located inside the app plugin `src` folder. They generally follow the same structure of a plugin and have their own `plugin.json`, but they don't have their own `package.json` or `.config` folder.
+
+Here's an example of a nested data source plugin:
+
+```diff bash
+./src
+ ├── README.md
+ ├── components
++├── nested-datasource
++│   ├── components
++│   │   ├── ConfigEditor.tsx
++│   │   └── QueryEditor.tsx
++│   ├── datasource.ts
++│   ├── img
++│   ├── module.ts
++│   ├── plugin.json
++│   └── types.ts
+ ├── img
+ │   └── logo.svg
+ ├── module.ts
+ └── plugin.json
+```
+
+:::note
+
+Notice that the plugin `nested-datasource` doesn't have a `package.json` of its own. The name of the nested plugin folder is not important.
+
+:::
+
+## How to add a nested plugin to an app plugin
+
+1. Generate the nested plugin:
```

Comment:
```suggestion
1. Create a new plugin that will become the nested plugin:
```

The phrasing "Generate the nested plugin" here seems confusing, because what you're actually doing is creating a brand new plugin that will later become a nested plugin.